### PR TITLE
Separate error code for insufficient storage

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs or @vekterli please review.

### CODE REVIEW

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java [hunk 1]
```diff
@@ -52,8 +52,9 @@ default void shutdown() { }
         NOT_FOUND,
         PRECONDITION_FAILED,
         BAD_REQUEST,
+        ERROR,
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java:55]: Reordering an enum should in practice not be a problem, but any usage of `.ordinal()` may result in unexpected behaviour.

**Review by bjorncs — APPROVED** [review]

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java (hunk 1 above)
Comment (jonmv) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java:55]: Which is why the use of `ordinal()` is discouraged, and definitely not used for this (brand new) enum :)

**Comment by vekterli** [comment]
Fails Travis build with
```
[ERROR] Failed to execute goal com.yahoo.vespa:abi-check-plugin:7-SNAPSHOT:abicheck (default) on project documentapi: ABI spec mismatch. To update run 'mvn package -Dabicheck.writeSpec' -> [Help 1]
```

**Comment by jonmv** [comment]
https://github.com/vespa-engine/vespa/pull/14716
`@PublicApi` :)))))))

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java (hunk 1 above)
Comment (bjorncs) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/DocumentOperationExecutor.java:55]: Both `.name()` and `.ordinal()` is bad practice, yet we do it (at least the former) in our own code ;)

**Pull request merged**